lxd/migrate_storage_volumes.go

```diff
@@ -365,8 +365,13 @@ func (c *migrationSink) DoStorage(state *state.State, projectName string, poolNa
 			_, targetSnapName, _ := api.GetParentAndSnapshotName(targetSnap.Name)
 
 			targetSnapshotsComparable = append(targetSnapshotsComparable, storagePools.ComparableSnapshot{
-				Name:         targetSnapName,
-				CreationDate: targetSnap.CreationDate,
+				Name: targetSnapName,
+
+				// The list of source snapshots from the offer header
+				// contains the creation timestamps in seconds granularity.
+				// Also use second based granularity for the target snapshots to be able to compare them.
+				// They are stored with nanoseconds in the database.
+				CreationDate: targetSnap.CreationDate.Round(time.Second),
```

Comment:
Hrm, do you think we may get problems with incorrect comparisons (triggering unnecessary refreshes) if the time.Unix() conversion used on the sending side does a different rounding (or non at all) to that which time.Round() uses here?

Perhaps we should convert to unix timestamp using time.Unix() and then back into a timestamp?